xdsl/pattern_rewriter.py

```diff
@@ -142,12 +142,13 @@ def replace_all_uses_with(
         self, from_: SSAValue, to: SSAValue | None, safe_erase: bool = True
     ):
         """Replace all uses of an SSA value with another SSA value."""
-        for use in from_.uses:
-            self.handle_operation_modification(use.operation)
+        modified_ops = [use.operation for use in from_.uses]
```

Comment:
Did you meant to use a set here, to instead only call the function once per operation?